scripts/biomedical/Human-GEM_Human1/scripts/format_metabolites.py

```diff
@@ -0,0 +1,466 @@
+        df.loc[df["hmdb"] == res["?hmdb"], "chembl"]\
+                        = res["?chembl"].split("/")[1]
+    print("DONE converting hmdb to chembl with data common query")
+
+
+def fill_chembl_from_pubchem(df):
+    """convert all chebi to pubchem with
+    Data Commons query and fill missing
+    chembl with converted results inplace
+    Args:
+        df: pandas.Dataframe with chemble and chebi info
+    Returs:
+        None
+    """
+    pub_chem_list = df[df["chembl"].isna()]\
+        ["pubchem.compound"].unique()[1:].astype(int).astype(str)
+    result = query_chembl_from_pubchem(pub_chem_list)
+    for res in result:
+        df.loc[df["pubchem.compound"] == res["?pubChem"], "chembl"]\
+                         = res["?chembl"].split("/")[1]
+    print("DONE converting pubchem to chembl with data common query")
+
+
+def fill_chembl_from_kegg(df):
+    """convert all chebi to kegg with
+   Data Commons queryand fill missing
+   chembl with converted results inplace
+   Args:
+       df: pandas.Dataframe with chemble and chebi info
+   Returs:
+       None
+   """
+    val_list = list(df[df["chembl"].isna()]\
+               ["kegg.compound"].unique())
+    if np.nan in val_list:
+        val_list.remove(np.nan)
+    kegg_list = np.array(val_list)
+    result = query_chembl_from_kegg(kegg_list)
+    for res in result:
+        df.loc[df["kegg.compound"] == res["?kegg"], "chembl"] \
+                       = res["?chembl"].split("/")[1]
+    print("DONE converting kegg to chembl with data common query")
+
+
+def fill_chembl_from_chebi(df):
+    """convert all chebi to chembl with
+    Data Commons query and fill missing
+    chembl with converted results inplace
+    Args:
+        df: pandas.Dataframe with chemble and chebi info
+    Returs:
+        None
+    """
+    val_list = list(df[df["chembl"].isna()]["chebi"].unique())
+    if np.nan in val_list:
+        val_list.remove(np.nan)
+    chebi_list = np.array(val_list)
+    result = query_chembl_from_chebi(chebi_list)
+    for res in result:
+        df.loc[df["chebi"] == res["?chebi"], "chembl"]\
+                         = res["?chembl"].split("/")[1]
+    print("DONE converting chebi to chembl with data common query")
+
+
+def bioservice_kegg_to_chembl(df, uni):
+    """convert exisitn kegg id to chembl id
+    and fill in (in-place) missing chemble with bioservices
+    Args:
+        df: pd.DataFrame with kegg and chembl columns
+        uni: bioservice mapping object
+    Returns:
+        None
+    """
+    from_, to_ = "kegg.compound", "chembl"
+    from_list = list(df[df[to_].isna()][from_].unique())
+    if np.nan in from_list:
+        from_list.remove(np.nan)
+    mapping = uni.get_mapping("kegg_ligand", "chembl")
+    for val in from_list:
+        try:
+            df.loc[df[from_] == val, "chembl"] = mapping[val]
+        except KeyError:
+            pass
+    print("DONE converting kegg to chembl with boservices")
+
+
+def bioservice_chebi_to_chembl(df, uni):
```

Comment:
this method is very similar to bioservice_kegg_to_chembl and can just be one method that also takes from_ and to_ as arguments.